mock/mock_proxy.go

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/clearcontainers/proxy/api"
+	"github.com/stretchr/testify/assert"
+)
+
+const testContainerid = "123456789"
+const testToken = "testtoken"
```

Comment:
I have used an helper function in the Proxy to get the token. I am going to be dropping the testContainerId altogether I think, as I thought the shim may need the containerId when I started out. I dont think the shim needs to be passed the containerId at all.